src/ws_correction.py

```diff
@@ -211,39 +217,44 @@ def move_wrong_scheme_messages(user, data, coda_input_dir):
                     if len(plan_updates) > 0:
                         flattened_survey_updates[plan.raw_field] = "; ".join([u.message for u in plan_updates])
                         flattened_survey_updates[plan.time_field] = sorted([u.timestamp for u in plan_updates])[0]
-                        flattened_survey_updates[f"{plan.raw_field}_source"] = "; ".join([u.source for u in plan_updates])
+                        flattened_survey_updates[f"{plan.raw_field}_source"] = "; ".join(
+                            [u.source_field for u in plan_updates])
                     else:
                         flattened_survey_updates[plan.raw_field] = None
                         flattened_survey_updates[plan.time_field] = None
                         flattened_survey_updates[f"{plan.raw_field}_source"] = None
 
-            # Hide the survey keys currently in the TracedData which have had data moved away.
-            td.hide_keys({k for k, v in flattened_survey_updates.items() if v is None}.intersection(td.keys()),
-                         Metadata(user, Metadata.get_call_location(), time.time()))
+            # For each RQA message, create a copy of its source td, append the updated TracedData, and add this to
+            # the list of TracedData to be returned
+            raw_field_to_rqa_plan_map = {plan.raw_field: plan for plan in PipelineConfiguration.RQA_CODING_PLANS}
+            for target_field, update in rqa_updates:
+                corrected_td = update.source_td.copy()
```

Comment:
what's the diff in memory usage?

Reply:
Good question. We're copying the same number of objects as before, so we should only see a change if the copied objects are somehow now larger. I think this is the case, because traced data copies are shallow and there's now less overlap in the objects we're copying, but the overall increase should be small because the size of the TracedData this early in the pipeline are still relatively small.

I'm happy to instrument a couple of runs tonight when I'm not using my laptop for other things (but I don't think there's reason to hold up merging given the current smallish size of this pipeline and the benefits we get from having correct history)